### PR TITLE
Added support for nullable type conversion in TestCase

### LINKED ISSUE #814: Support nullable types with TestCase
For convenience it'd be nice if TestCase supported nullable types so you could have for example a DateTime? parameter instead of having to use TestCaseSource. 

```csharp
[TestCase("2000-01-01")]
public void Foo(DateTime? date)
{ }
```

### PR DESCRIPTION
**Pull request by rprouse**
Fixes #814

### CODE REVIEW

File: src/NUnitFramework/framework/Attributes/TestCaseAttribute.cs [hunk 1]
```diff
@@ -368,17 +368,34 @@ private static void PerformSpecialConversions(object[] arglist, IParameterInfo[]
...
                 if (convert)
-                    arglist[i] = Convert.ChangeType(arg, targetType, System.Globalization.CultureInfo.InvariantCulture);
+                {
+                    Type convertTo = targetType.IsGenericType && targetType.GetGenericTypeDefinition() == typeof(Nullable<>) ? 
+                        targetType.GetGenericArguments()[0] : targetType;
+                    arglist[i] = Convert.ChangeType(arg, convertTo, System.Globalization.CultureInfo.InvariantCulture);
+                }
+
+                // Convert.ChangeType doesn't work for TimeSpan from string
+                if ((targetType == typeof(TimeSpan) || targetType == typeof(TimeSpan?)) && arg is string)
```
Comment (rprouse) [inline src/NUnitFramework/framework/Attributes/TestCaseAttribute.cs]: We had a case for TimeSpan above, but when I added a test, it didn't pass

File: src/NUnitFramework/tests/Attributes/TestCaseAttributeTests.cs [hunk 2]
```diff
@@ -186,25 +192,6 @@ public void HandlesParamsArrayWithOneItemAsLastArgument(string s1, string s2, pa
             Assert.AreEqual("c", array[0]);
         }
 
-        [TestCase(1)]
```
Comment (rprouse) [inline src/NUnitFramework/tests/Attributes/TestCaseAttributeTests.cs]: These test cases were moved down with the rest of the nullable tests.

**Comment by rprouse** [comment]
Build failures are unrelated. I have restarted the failed builds for the push.

**Comment by oznetmaster** [comment]
This will not build under CF.

It references TimeSpan.TryParse which is not available in CF

**Pull request merged**